dashboard/app/controllers/admin_nps_controller.rb

```diff
@@ -0,0 +1,15 @@
+require 'cdo/env'
+require 'set'
+
+# The controller for manipulating NPS surveys
+class AdminNpsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_admin
+  check_authorization
+
+  # PUT /admin/nps_update
+  def nps_update
+    audience = params[:audience]
```

Comment:
Should we verify that the passed in param is one of the expected values?

Reply:
Added a check and a corresponding flash[alert], and a test for this case. This also inspired me to add a flash[notice] (and a corresponding test) for the user that confirms when the flag is updated correctly!